src/main/java/org/testng/reporters/SuiteHTMLReporter.java

```diff
@@ -312,15 +311,12 @@ private void generateMethodsChronologically(XmlSuite xmlSuite, ISuite suite,
       Collection<IInvokedMethod> invokedMethods = suite.getAllInvokedMethods();
       if (alphabetical) {
 	@SuppressWarnings({"unchecked"})
-	Comparator<? super ITestNGMethod>  alphabeticalComparator = new Comparator(){
-	  @Override
-	  public int compare(Object o1, Object o2) {
-	    IInvokedMethod m1 = (IInvokedMethod) o1;
-	    IInvokedMethod m2 = (IInvokedMethod) o2;
-	    return m1.getTestMethod().getMethodName().compareTo(m2.getTestMethod().getMethodName());
-	  }
-	};
-	Collections.sort((List) invokedMethods, alphabeticalComparator);
+	Comparator<? super ITestNGMethod>  alphabeticalComparator = (o1, o2) -> {
+    IInvokedMethod m1 = (IInvokedMethod) o1;
```

Comment:
As `invokedMethods` contains `IInvokedMethod`, I don't catch why we need to cast here.

Reply:
fixed